File: test-fixtures/cross-language/zig/realistic.zig
// SPDX-License-Identifier: MIT
const SessionState = struct {
    mutableField: []const u8,
    immutableField: []const u8,
    sharedName: usize,
    pub const staticField: usize = 1;
};

const AuditState = struct {
    sharedName: usize,
};

pub fn readSession(state: SessionState) usize {
    return state.sharedName + SessionState.staticField;
}

File: test-fixtures/cross-language/rust/realistic.rs
// SPDX-License-Identifier: MIT
pub struct SessionState<T> {
    pub mutable_field: T,
    pub shared_name: usize,
    private_field: bool,
}

pub struct AuditState {
    pub shared_name: usize,
}

impl<T> SessionState<T> {
    pub fn new(value: T) -> Self {
        Self {
            mutable_field: value,
            shared_name: 0,
            private_field: false,
        }
    }
}
